### PR TITLE
VB-729: Tidy Cypress config and mocks

### PR DESCRIPTION
**Pull request by tpmcgowan**
Preparatory work before adding new tests for booking journey:
* tidied Cypress config (make consistent, cleanup following automated migration from v9 to v10)
* mock APIs
  * added types
  * updated test data to reflect schema changes
  * removed deprecated endpoints

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -23,57 +20,37 @@ export default defineConfig({
...
-        }: {
-          results: { totalPages: number; totalElements: number; content: Partial<Prisoner>[] }
-          page: string
-          size: string
-        }) => offenderSearch.getPrisoners(results, page, size),
-        stubGetPrisoner: (results: { totalPages: number; totalElements: number; content: Partial<Prisoner>[] }) =>
-          offenderSearch.getPrisoner(results),
+
+        // Prisoner Contact Registry
+        stubGetPrisonerSocialContacts: prisonerContactRegistry.stubGetPrisonerSocialContacts,
```
Comment (tpmcgowan) [inline cypress.config.ts:33]: Have simplified this config from what was auto-generated during upgrade from v9 to v10 (having looked at what other projects are doing...)

File: integration_tests/mockApis/offenderSearch.ts [hunk 2]
```diff
@@ -3,14 +3,14 @@ import { stubFor } from './wiremock'
 import { Prisoner } from '../../server/data/prisonerOffenderSearchTypes'
 
 export default {
-  getPrisoners: (
+  stubGetPrisoners: (
```
Comment (tpmcgowan) [inline integration_tests/mockApis/offenderSearch.ts:6]: have named all these mock APIs to start with `stub` as we had a mix and this looks to be consistent with other projects

File: integration_tests/mockApis/wiremock.ts [hunk 3]
```diff
@@ -10,20 +10,4 @@ const getMatchingRequests = body => superagent.post(`${url}/requests/find`).send
 const resetStubs = (): Promise<Array<Response>> =>
   Promise.all([superagent.delete(`${url}/mappings`), superagent.delete(`${url}/requests`)])
 
-const postFor = ({ body, urlPattern, urlPath }) =>
```
Comment (tpmcgowan) [inline integration_tests/mockApis/wiremock.ts:13]: Didn't seem to be used. Can add back if needed...

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**